src/Dropdown/MultiDropdownWithSelectAll.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+import Select, { components } from "react-select";
+
+import Creatable from "react-select/lib/Creatable";
+import { Props } from "react-select/lib/Creatable";
+import * as cx from "classnames";
+import {
+  CommonProps,
+  defaultProps,
+  defaultComponents,
+  defaultStyle,
+  getCommonClassnames,
+  DefaultProps
+} from "./commons";
+import { GroupType, ValueType } from "react-select/lib/types";
+
+export function allSelected<OptionType = never>(): SelectAllValue<OptionType> {
+  return { type: "AllSelected" };
+}
+
+export function someSelected<OptionType>(values: OptionType[]) {
+  return { type: "SomeSelecetd", values };
+}
+
+type SelectAllOptionType = { value: "_ALL"; label: string };
+type SelectAllValue<OptionType> =
+  | {
+      type: "AllSelected";
+    }
+  | {
+      type: "SomeSelected";
+      values: OptionType[];
+    };
+
+type NonDefaultProps<OptionType> = CommonProps<
```

Comment:
does `CommonProps` really need the `OptionType` type param? Meaning, shouldn't all value/option related props be handled specifically by each variant outside of `CommonProps`? If it's only used internally to pass it to react select Props type, maybe we can use `unknown` there

Reply:
There are many other props, like `components` or `options` that are not handled by specific cases. I tried to remove the type param, but I have many errors when I pass these props to react-select

Reply:
if `options` is part of `CommonProps` then it should be `options: OptionType[]` i.e. not include `SelectAllOptionType` externally. Right?

Reply:
Yep, I noticed this few minutes ago and fixed it ;)